compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -143,7 +194,7 @@ if (secondChildNeedsRelocation)
              * We need to use a b.cond instruction instead for that case.
              */
             && ((node->getNumChildren() != 3) ||
-             (node->getChild(2)->getNumChildren() != cg->getLinkage()->getProperties().getNumAllocatableIntegerRegisters())))
+             (getNumberOfIntegerNodesInGlRegDeps(node->getChild(2), cg) < cg->getLinkage()->getProperties().getNumAllocatableIntegerRegisters())))
```

Comment:
If `src1Reg` is not part of the GlRegDeps, does it need to be accounted for in this check if you will be eventually adding it to the reg dep below?

Reply:
When this check passes, the number of integer child nodes of the GlRegDeps is less than the maximum allocatable integer registers (25 in our case), so even if `src1Reg` is not part of the GlRegDeps, the number of integer registers added to the reg dep does not exceed the limit (25).
If `src1Reg` is part of the GlRegDeps, the number of integer children can be less than or equal to the limit, but to check if `src1Reg` is part of the GlRegDeps, we need to evaluate the GlRegDeps node first.